.github/workflows/build.yml

```diff
@@ -42,6 +42,20 @@ jobs:
         ./run_example.sh simple_app show_env
         ./run_example.sh simple_app_under_oe show_env
 
+        #! ---------------------------------------------------------------------
+        #! Exercise various interfaces in --dry-run mode. This will ensure that
+        #! script's execution logic will likely work for different sample apps,
+        #! when tested on the appropriate platform and environment.
+        #! ---------------------------------------------------------------------
+        ./run_example.sh --dry-run simple_app
+        ./run_example.sh --dry-run simple_app setup
+        ./run_example.sh --dry-run simple_app run_test
+
+        ./run_example.sh --dry-run simple_app_under_oe
+        ./run_example.sh --dry-run simple_app_under_oe setup
+        ./run_example.sh --dry-run simple_app_under_oe run_test
+        ./run_example.sh --dry-run simple_app_under_oe setup_with_auto_policy_generation_for_OE
```

Comment:
Added few variations to CI to ensure that future script changes for other apps don't regress.

This is not fool-proof testing but goes a long ways to ensure some regression-checking  for code changes affecting apps on other platforms.